src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeTypeBuilder.cs

```diff
@@ -1627,11 +1611,14 @@ protected override TypeInfo CreateTypeInfoCore()
 
                     body = meth.GetBody();
```

Comment:
This can be moved right after `CreateMethodBodyHelper`. If we did not end up calling `CreateMethodBodyHelper`, no need to refetch the `body`.